tasks/kibana.yml

```diff
@@ -1,20 +1,36 @@
 ---
-- name: Download Kibana tarball.
-  get_url:
-    url: https://download.elastic.co/kibana/kibana/kibana-{{ elk_kibana_version }}-linux-x64.tar.gz
-    dest: /opt/kibana-{{ elk_kibana_version }}-linux-x64.tar.gz
-    # Don't download file every time, only if it doesn't exist.
-    force: no
+- name: Add Kibana apt repo.
+  apt_repository:
+    repo: "deb http://packages.elastic.co/kibana/4.5/debian stable main"
+    state: present
+  notify:
+    - reload systemd
 
-- name: Check for existing Kibana installation.
+  # Previous versions of the role used a symlink to support tarball extraction.
+  # Let's provide a sane upgrade path and handle the edge case where it's still
+  # a symlink by converting it to a directory.
+- name: Check Kibana directory.
   stat:
-    path: /opt/kibana-{{ elk_kibana_version }}-linux-x64
+    path: /opt/kibana
   register: kibana_directory_check
 
-  # It's important to configure the init file first, since it notifies daemon-reload.
-  # This task must come before any task in the same play that will notify `restart kibana`,
-  # since Ansible queues handlers in the order they were notified. We want to ensure that if
-  # kibana is being restarted AND the service file changed, daemon-reload runs first.
+- name: Remove Kibana symlink.
+  file:
+    path: /opt/kibana
+    state: absent
+  when: kibana_directory_check.stat.islnk
```

Comment:
I went ahead and included handling of the old symlink at /opt/kibana. Cleaning up the old kibana tarballs seems out of scope for the role, but during testing of an upgrade from the symlink approach to the apt repo, apt blithely followed the symlink and dumped all the new version's files into the old version's directory. It was a mess. This change handles that edge case and keeps the role idempotent.